packages/contracts/contracts/universal/MultiSigWallet.sol

```diff
@@ -0,0 +1,529 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "./IMultiSigWallet.sol";
+import { Initializable } from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+
+/**
+ * @custom:proxied
+ * @title MultiSigWallet
+ * @notice Allows multiple parties to agree on transactions before execution.
+ */
+contract MultiSigWallet is IMultiSigWallet, Initializable {
+    /**
+     * @notice A mapping of transactions submitted.
+     */
+    mapping (uint => Transaction) public transactions;
+
+    /**
+     * @notice A mapping of confirmations.
+     */
+    mapping (uint => mapping (address => bool)) public confirmations;
+
+    /**
+     * @notice A mapping that records owner status.
+     */
+    mapping (address => bool) public isOwner;
+
+    /**
+     * @notice A list of owners.
+     */
+    address[] public owners;
+
+    /**
+     * @notice A number of confirmations required to execute a transaction.
+     */
+    uint public numConfirmationsRequired;
+
+    /**
+     * @notice Number of transactions submitted.
+     */
+    uint public transactionCount;
+
+    /**
+     * @notice Represents transaction data.
+     *
+     * @custom:field destination The destination address to run the transaction.
+     * @custom:field value       The value passed in while executing the transaction.
+     * @custom:field data        Calldata for transaction.
+     * @custom:field executed    Record whether a transaction was executed or not.
+     */
+    struct Transaction {
+        address destination;
+        uint value;
+        bytes data;
+        bool executed;
```

Comment:
Please place `executed` right below the `destination` for slot optimization.

Reply:
Good point. I'll apply it.